packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +38,157 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+//get data from custom polcies file and write custom policies to CFN template
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceName: string,
+  service: string,
+  resourceDir: string,
+  cfnFile: string,
+  category: string,
+  filePath: string
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(service, category, resourceName);
+  if (!validateExistCustomPolicies(customPolicies)) {
+    return;
+  }
+  await validateCustomPoliciesSchema(customPolicies, category, resourceName);
+
+  await addCustomPoliciesToCFNTemplate(service, customPolicies.policies, cfnTemplate, filePath, resourceName, {templateFormat} );
+
+}
+
+//merge the custom IAM polciies to CFN template for lambda and API container
+
+async function addCustomPoliciesToCFNTemplate(
+  service: string,
+  customPolicies: CustomIAMPolicy[],
+  cfnTemplate: Template,
+  filePath: string,
+  resourceName: string,
+  {templateFormat}: any
+  ) {
+  let customExecutionPolicy;
+  const { envName } = stateManager.getLocalEnvInfo();
+  if(service === 'Lambda') {
+    customExecutionPolicy = customExecutionPolicyForFunction;
+  }
+  if (service === 'ElasticContainer') {
+    const roleName = cfnTemplate.Resources.TaskDefinition.Properties.ExecutionRoleArn["Fn::GetAtt"][0];
+    customExecutionPolicy = customExecutionPolicyForContainer;
+    const role = Fn.Ref(roleName);
+    customExecutionPolicy.Properties.Roles.push(role);
+  }
+
+  warnWildcardCustomPolicies(customPolicies, resourceName);
+
+  for (const customPolicy of customPolicies) {
+    validateCustomPolicy(customPolicy, resourceName);
+    const policyWithEnv = await replaceEnvForCustomPolicies(customPolicy, envName);
+    if (!policyWithEnv.Effect) policyWithEnv.Effect = 'Allow';
+    customExecutionPolicy.Properties.PolicyDocument.Statement.push(policyWithEnv);
+  }
+
+  if (service === 'Lambda') {
+    cfnTemplate.Resources.CustomLambdaExecutionPolicy = customExecutionPolicy;
+  }
+  if (service === 'ElasticContainer') {
+    cfnTemplate.Resources.CustomExecutionPolicyForContainer = customExecutionPolicy;
+  }
+
+  await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+}
+
+
+//validate the format of actions and ARNs for custom IAM policies
+function validateCustomPolicy(customPolicy: CustomIAMPolicy, resourceName: string) {
+  const resources = customPolicy.Resource;
+  const actions = customPolicy.Action;
+  let resourceRegex = new RegExp('arn:(aws[a-zA-Z0-9-]*):([a-zA-Z0-9\\-])+:([a-z]{2}(-gov)?-[a-z]+-\\d{1})?:(\\d{12})?:(.*)');
+  let actionRegex = new RegExp('[a-zA-Z0-9]+:[a-z|A-Z|0-9|*]+');
+  let wrongResourcesRegex = [];
+  let wrongActionsRegex = [];
+  let errorMessage = "";
+
+  for (const resource of resources) {
+    if (!resourceRegex.test(resource) && !(resource === '*')) {
```

Comment:
```suggestion
    if (!resourceRegex.test(resource) && resource !== '*') {
```

Reply:
In Attila's comment, !(.... === ) instead of !==